### ISSUE TITLE
Panel/Menu applet: Auto-hide panel partly occludes menu applet when opened with meta-key

### ISSUE THREAD

**Issue by fredcw** [linuxmint/cinnamon#10970]
```
 * Cinnamon version  5.4.8
 * Bug not present in Mint 20.3
 * Distribution - Mint 21 daily PPA and Manjaro
 ```

**Issue**
With panel option set to "Auto hide panel" or "Intelligently hide panel", opening the menu applet with the meta key (win key) results in the menu being partly occluded by the panel.

Edit: The menu option "Force the panel to be visible when opening the menu" must also be _on_

![Screenshot from 2022-07-27 03-07-37](https://user-images.githubusercontent.com/58893963/181192159-057bcc48-6d69-4d2b-b2b0-883114412170.png)

line 1465: `this.panel.peekPanel();` in `_onOpenStateChanged()` in menu applet not working as before.

**Comment by spamegg1** [comment]
I can confirm the issue on Cinnamon 5.4.8
![Untitled](https://user-images.githubusercontent.com/4255997/181822414-6464ec51-a374-457c-9e89-450ae0188619.png)

**Comment by nhamawi** [comment]
Latest Cinnamon is still an issue. The image below shows that the search box is even obscured when the panel is docked at the top of the screen:

![Screenshot from 2023-03-16 12-25-56](https://user-images.githubusercontent.com/2650762/225693104-714dc60f-5467-441f-a889-074642ad1a70.png)

**Comment by mydoghasfleas** [comment]
I am still getting this issue in Cinnamon 5.8.4.

When I set the panel to Auto or Intelligent hide and bring it up with the Super key, the panel overlaps the menu:

![image](https://github.com/linuxmint/cinnamon/assets/125179869/e39e6bc2-ae36-43a1-84ff-13124a9698b2)

**Comment by fredcw** [comment]
@mydoghasfleas This bug has since been fixed in cinnamon but not before the 5.8.4 release (Mint 21.2)
